opensir/models/model.py

```diff
@@ -175,10 +181,34 @@ def fit(self, t_obs, n_i_obs, population, fit_index=None):
             Model: Reference to self
         """
 
+        # If input not np.ndarray, cast
+        if not isinstance(t_obs, np.ndarray):
+            t_obs = np.array(t_obs)
+
+        if not isinstance(n_i_obs, np.ndarray):
+            n_i_obs = np.array(n_i_obs)
+
+        # Raise error if any time or number of infected is negative
+        if any(t_obs < 0) | any(n_i_obs < 0):
+            raise self.InvalidInputsForFittingError(Exception)
+
+        # Raise error if any time or number of infected lists
+        # is non monotonically increasing
+        if any(np.diff(t_obs) < 0) | any(np.diff(n_i_obs) < 0):
+            raise self.InvalidInputsForFittingError(Exception)
+
+        # Check consistent dimensions
+        if len(t_obs) != len(n_i_obs):
```

Comment:
Non blocking comment, but maybe this restriction could be dropped if we merge `t_obs` and `n_i_obs` together.
In practice this data comes from a CSV file (unless someone wants to write them by hand, which might not be practical).
By default, np.array will complain it's not a matrix (non rectangular or non square)

Reply:
Let's keep these guys separated as in many cases the number of observed and the datetime index require different pre-processing.